windows/winrm-gen.ps1

```diff
@@ -0,0 +1,83 @@
+param (
+    [string]$ComputerName = "servername.domain.com"
+)
+
+$FirewallParam = @{ 
+    DisplayName = 'Windows Remote Management (HTTPS-In)'
+    Direction = 'Inbound' 
+    LocalPort = 5986 
+    Protocol = 'TCP' 
+    Action = 'Allow' 
+    Program = 'System' 
+} 
+
+$ErrorActionPreference = "Stop"
+
+#
+# WinRM prechecks
+#
+try {
+    Write-Output "Checking if WinRM is enabled"
+    Test-WSMan > $null
+    Write-Output "Checking if WinRM configured"
+    winrm get winrm/config
+    if ( $LASTEXITCODE ) {
+        Write-Output "WinRM not configured, configuring with the default settings"
+        winrm quickconfig -transport:https
+    }
+}
+catch [System.Management.Automation.CommandNotFoundException] {
+    "winrm not found"
+    exit
+}
+catch {
+    "An error occurred that could not be resolved."
+    exit
+}
+
+Write-Output "Setting the auth service to Basic"
+winrm set winrm/config/service/auth '@{Basic="true"}'
+if ( $LASTEXITCODE ) {
+    Write-Output "Warning: the auth service could not be set"
+}
+
+# Validate the service settings /// winrm set already gives the output.
+# Write-Output "Checking the auth service was set"
+# winrm get winrm/config/service/Auth
+
+Write-Output "Generating new SSL Certificate"
+$Cert = New-SelfSignedCertificate -Subject "CN=`"$ComputerName`"" -TextExtension '2.5.29.37={text}1.3.6.1.5.5.7.3.1'
```

Comment:
The only problem here is that the generated self-signed certificate validity defaults to 1 year. To should not suffice in our context, if we wish to use the same Windows worker for longer periods. Let's set it to 5 or even 10 years:

```powershell
$fiveYearsValidity = $(get-date).ToUniversalTime().AddYears(5)
$Cert = New-SelfSignedCertificate -Subject "CN=`"$ComputerName`"" -TextExtension '2.5.29.37={text}1.3.6.1.5.5.7.3.1' -NotAfter $fiveYearsValidity
```